lib/chef/resource/aws_internet_gateway.rb

```diff
@@ -1,16 +1,43 @@
-require 'chef/provisioning/aws_driver/aws_resource'
+require 'chef/provisioning/aws_driver/aws_resource_with_entry'
 
-class Chef::Resource::AwsInternetGateway < Chef::Provisioning::AWSDriver::AWSResource
-  aws_sdk_type AWS::EC2::InternetGateway, load_provider: false, id: :id
+#
+# An AWS internet gateway, allowing communication between instances inside a VPC and the internet.
+#
+# `name` is not guaranteed unique for an AWS account; therefore, Chef will
+# store the internet gateway ID associated with this name in your Chef server in the
+# data bag `data/aws_internet_gateway/<name>`.
+#
+# API documentation for the AWS Ruby SDK for VPCs (and the object returned from `aws_object` can be found here:
+#
+# - http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/EC2/InternetGateway.html
+#
+class Chef::Resource::AwsInternetGateway < Chef::Provisioning::AWSDriver::AWSResourceWithEntry
+  aws_sdk_type AWS::EC2::InternetGateway
 
+  require 'chef/resource/aws_vpc'
```

Comment:
You shouldn't need this require line - was it giving you an error if you didn't?